pdns/slavecommunicator.cc

```diff
@@ -91,8 +91,12 @@ void CommunicatorClass::ixfrSuck(const DNSName &domain, const TSIGTriplet& tt, c
   try {
     DNSSECKeeper dk (&B); // reuse our UeberBackend copy for DNSSECKeeper
 
-    if(!B.getDomainInfo(domain, di) || !di.backend || di.kind != DomainInfo::Slave) { // di.backend and B are mostly identical
-      g_log<<Logger::Error<<"Can't determine backend for domain '"<<domain<<"'"<<endl;
+    bool wrongDomainKind = false;
+    if(!B.getDomainInfo(domain, di) || !di.backend || (wrongDomainKind = true, di.kind != DomainInfo::Slave)) { // di.backend and B are mostly identical
```

Comment:
This looks super clever and I cannot read it. Can we change it? Maybe some `()` ?

Reply:
if this is problematic, a few more ()'s are not going to help I'm afraid - unless I misunderstand where your confusion is coming from. I can also rewrite this into 7 lines of not much more obvious code. The way to read the above is that it sets a flag if it hit the third || component. Let me know!

Reply:
The () would help if `(wrongDomainKind = true, di.kind != DomainInfo::Slave)` meant `(wrongDomainKind = (true, di.kind != DomainInfo::Slave))` but it does not, so that's okay.

So, if we hit the third component, `wrongDomainKind` is set to true, even if the third component then evaluates to false. But this is fine because we then only look at `wrongDomainKind` if the third component was true. I get it now, thanks!

Reply:
I'll add a comment.